rules/manage-bundle-size/rule.md

```diff
@@ -0,0 +1,57 @@
+---
+type: rule
+archivedreason: 
+title: Do you know the best ways to manage your bundle size?
+guid: 0248bded-7da4-4dfe-8151-d4ac1b9559a8
+uri: manage-bundle-size
+created: 2023-07-14T02:13:28Z
+authors:
+- title: Harry Ross
+  url: https://www.ssw.com.au/people/harry-ross
+related:
+- the-best-package-manager-for-react
+- monitor-packages-for-vulnerability
+- packages-up-to-date
+---
+
+
+Shipping a large JavaScript bundle can be catastrophic for your web app - resulting in longer load times and a poor user experience using a web app. It is very easy for this to be the case when using a web framework such as React or Angular, as adding packages to an already large amount of framework code can result in a bloated JavaScript bundle. 
+
+<!--endintro-->
+
+Tools like [Webpack](https://webpack.js.org/), [Rollup](https://rollupjs.org/) or [Vite](https://vitejs.dev/) can help with reducing the amount of JavaScript shipped to the end-user. However, even if we have used these tools it may still be necessary to minimise the number of NPM packages in the project.
+
+## depcheck
+
+Depcheck is a great CLI tool that scans your project to automatically detect unused dependencies in a project, giving you a clear and concise list of unused packages in a repository. 
+
+```shell
+npx depcheck
+```
+
+![Figure: A list of the unused dependencies in a project](depcheck.png)
+
+## Bundlephobia
+
+Bundlephobia is a great tool for quickly seeing the bundle size of NPM packages. You can find the bundle size of any NPM package by going to the URL: bundlephobia.com/package/{{ PACKAGE_NAME_HERE }}. 
+
+You can also upload a package.json file to display a list of packages in your project, allowing you to easily see which NPM packages have the largest size in the project. You can find it at [bundlephobia.com/scan](https://bundlephobia.com/scan).
+
+![Figure: The list of packages from the package.json file, sorted by size](bundlephobia_list.png)
+
+## Import Cost VSCode Extension
+
+The Import Cost extension for VSCode, developed by Wix shows the size of each NPM package you import, which can provide useful visual feedback on how large the packages you importing actually are. You can find the extension [here](https://marketplace.visualstudio.com/items?itemName=wix.vscode-import-cost). 
```

Comment:
Hey @Harry-Ross 

1. Can you follow https://www.ssw.com.au/rules/descriptive-links/ for the link in this sentence?